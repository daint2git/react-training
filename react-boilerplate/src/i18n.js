import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'

import en from 'translations/en'
import ja from 'translations/ja'

i18n
  .use(initReactI18next) // pass the i18n instance to react-i18next
  .init({
    resources: {
      en: { translation: en },
      ja: { translation: ja },
    },
    lng: 'ja',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false, // not needed for react as it escapes by default
    },
  })

export default i18n
