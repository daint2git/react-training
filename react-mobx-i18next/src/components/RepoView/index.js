import React from 'react'
import { inject, observer } from 'mobx-react'
import sum from 'lodash/sum'
import { withTranslation, useTranslation } from 'react-i18next'
import RepoContainer from '@components/RepoContainer'
import RepoList from '@components/RepoList'

@inject('RepoStore')
@observer
class RepoView extends React.Component {
  componentDidMount() {
    const {
      RepoStore: { fetchRepos },
    } = this.props

    fetchRepos()
  }

  changeLanguage = lng => {
    const { i18n } = this.props
    i18n.changeLanguage(lng)
  }

  render() {
    const {
      RepoStore: { repos, repoCount },
      t,
    } = this.props

    console.log('withTranslation', withTranslation, useTranslation, this.props.i18n)

    return (
      <RepoContainer className={`container`}>
        <h3>Repos</h3>
        <h4>{sum([1, 2, 3])}</h4>
        <RepoList repos={repos} repoCount={repoCount} />
        <br />
        <p>{t('description.part1')}</p>
        <p>{t('description.part2')}</p>
        <div>
          <button onClick={() => this.changeLanguage('vi')}>vi</button>
          <button onClick={() => this.changeLanguage('en')}>en</button>
        </div>
      </RepoContainer>
    )
  }
}

export default withTranslation()(RepoView)
