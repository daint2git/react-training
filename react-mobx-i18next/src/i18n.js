import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import en from './translations/en'
import vi from './translations/vi'

i18n
  .use(initReactI18next) // pass the i18n instance to react-i18next
  .init({
    resources: {
      en: { translation: en },
      vi: { translation: vi },
    },
    lng: 'vi',
    fallbackLng: 'vi',
    interpolation: {
      escapeValue: false, // not needed for react as it escapes by default
    },
  })

export default i18n
