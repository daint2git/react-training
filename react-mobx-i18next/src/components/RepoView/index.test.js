import RepoView from '../RepoView'
import i18n from '../../i18n'

const renderComponentWithStore = store => mount(<RepoView RepoStore={store} i18n={i18n} />)

describe('RepoView', () => {
  let wrapper

  afterEach(() => {
    wrapper.unmount()
  })

  test('should render correctly default data', () => {
    const mockStore = {
      // observable
      repos: [],
      repoCount: 0,
      // actions
      fetchRepos: jest.fn(),
    }
    wrapper = renderComponentWithStore(mockStore)
    expect(wrapper).toMatchSnapshot()
  })

  test('should render correctly mock data', () => {
    const repos = [
      {
        id: 1,
        htmlUrl: 'abc1',
      },
      {
        id: 2,
        htmlUrl: 'abc2',
      },
    ]

    const mockStore = {
      // observable
      repos,
      repoCount: repos.length,
      // actions
      fetchRepos: jest.fn(),
    }

    wrapper = renderComponentWithStore(mockStore)
    expect(wrapper).toMatchSnapshot()
  })

  test('should call fetchRepos callback', () => {
    const mockStore = {
      // observable
      repos: [],
      repoCount: 0,
      // actions
      fetchRepos: jest.fn(),
    }
    wrapper = renderComponentWithStore(mockStore)
    expect(mockStore.fetchRepos).toHaveBeenCalled()
    expect(mockStore.fetchRepos).toHaveBeenCalledTimes(1)
  })
})
