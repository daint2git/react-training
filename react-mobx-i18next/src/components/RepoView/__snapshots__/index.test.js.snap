// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoView should render correctly default data 1`] = `
<div
  className="container"
>
  <h2>
    t('title')
  </h2>
  <h3>
    Repos
  </h3>
  <h4>
    6
  </h4>
  Array [
    <ul />,
    <p>
      Have 
      0
       repos.
    </p>,
  ]
  <br />
  <p>
    t('description.part1')
  </p>
  <p>
    t('description.part2')
  </p>
  <div>
    <button
      onClick={[Function]}
    >
      vi
    </button>
    <button
      onClick={[Function]}
    >
      en
    </button>
  </div>
</div>
`;

exports[`RepoView should render correctly mock data 1`] = `
<div
  className="container"
>
  <h2>
    t('title')
  </h2>
  <h3>
    Repos
  </h3>
  <h4>
    6
  </h4>
  Array [
    <ul>
      <li
        key="1"
      >
        abc1
      </li>
      <li
        key="2"
      >
        abc2
      </li>
    </ul>,
    <p>
      Have 
      2
       repos.
    </p>,
  ]
  <br />
  <p>
    t('description.part1')
  </p>
  <p>
    t('description.part2')
  </p>
  <div>
    <button
      onClick={[Function]}
    >
      vi
    </button>
    <button
      onClick={[Function]}
    >
      en
    </button>
  </div>
</div>
`;
